### PR TITLE
Restructure permissions

### PR DESCRIPTION
**Pull request by prinsniels**
Fixed permission settings
-> Export file now runs via email and had correct permissions
-> Location recording after boot does not start when the permission is revoked or not given
-> Location permission request deny does no longer star the recording service

### CODE REVIEW

File: app/src/main/java/nl/nfi/cellscanner/BootDeviceReceiver.java [hunk 1]
```diff
@@ -29,7 +29,8 @@ public void onReceive(Context context, Intent intent) {
           is inRecordingState
          */
         if (Recorder.inRecordingState(context)){
-            Recorder.startService(context);
+            if (PermissionSupport.hasAccessCourseLocationPermission(context)) Recorder.startService(context);
+            else Recorder.setRecordingState(context, false);
```
Comment (jokkalt) [inline app/src/main/java/nl/nfi/cellscanner/BootDeviceReceiver.java:33]: Ik ben voor consistent wel-haakjes gebruik in if-else

File: app/src/main/java/nl/nfi/cellscanner/Database.java [hunk 2]
```diff
@@ -33,7 +32,12 @@
     private SQLiteDatabase db;
     private Date previous_date = null;
 
-    public static File getDataPath(Context ctx) {
+    public static String getFileTitle() {
```
Comment (jokkalt) [inline app/src/main/java/nl/nfi/cellscanner/Database.java:35]: generateFileTitle is misschien beter; het bestand bestaat nog niet op dit punt als ik het goed zie

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context=".MainActivity">
+    tools:context="nl.nfi.cellscanner.MainActivity">
```
Comment (jokkalt) [inline app/src/main/res/layout/activity_main.xml:7]: This wasn't explicit before; why the change?

**Review by jokkalt — APPROVED** [review]
Works on my phone

File: app/src/main/java/nl/nfi/cellscanner/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSION_REQUEST_START_RECORDING: {
+                if (permissions.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    requestStartRecording();
+                } else {
+                    // explain the app will not be working
+                    Toast.makeText(this,"App will not work without location permissions", Toast.LENGTH_SHORT).show();
+                    toggleButtonsRecordingState();
```
Comment (jokkalt) [inline app/src/main/java/nl/nfi/cellscanner/MainActivity.java:106]: Would be convenient to have a new location access requester here; currently will just tell me it doesn't have permissions

**Comment by jokkalt** [comment]
- [x] This should be merged in the Feature/post api lvl 26 branch I think?

File: app/src/main/java/nl/nfi/cellscanner/MainActivity.java (hunk 4 above)
Comment (prinsniels) [reply app/src/main/java/nl/nfi/cellscanner/MainActivity.java:106]: This is the callback of the request for permission. According to best practice we should not recall here. =D Otherwise we hit an infinite loop

https://developer.android.com/training/permissions/requesting
